src/main/java/org/spongepowered/common/service/scheduler/AsyncScheduler.java

```diff
@@ -106,19 +51,20 @@
     private final Condition condition = this.lock.newCondition();
     // The dynamic thread pooling executor of asynchronous tasks.
     private ExecutorService executor;
-    // Query actor for task information
-    private SchedulerHelper schedulerHelper;
 
     private AsyncScheduler() {
-        this.schedulerHelper = new SchedulerHelper(ScheduledTask.TaskSynchroncity.ASYNCHRONOUS);
+        super(ScheduledTask.TaskSynchroncity.ASYNCHRONOUS);
 
-        new Thread(new Runnable() {
+        Thread thread = new Thread(new Runnable() {
 
             @Override
             public void run() {
                 stateMachineBody();
             }
-        }).start();
+        });
+        thread.setName("Sponge Async Scheduler Thread");
+        thread.setDaemon(true);
+        thread.start();
     }
 
     private void stateMachineBody() {
```

Comment:
Since your changes, this name has lost a lot of its meaning and is probably best being called a loop of some sort